Chore/renamed gp module

kcw-grunt opened the pull request:
## 📱 Description
<!-- Provide a brief description of what this PR accomplishes -->

## Platform
- [x] Android
- [ ] iOS
- [x] Games-Unity
- [ ] DevOps (AWS)
- [ ] Website
- [ ] C/Golang 

## 🎯 Type of Change
<!-- Mark the relevant option with an [x] -->
- [ ] 🐛 Bug fix
- [ ] ✨ New feature
- [ ] 🔧 Refactoring (code change that neither fixes a bug nor adds a feature)
- [ ] 📚 Documentation update
- [ ] 🎨 UI/UX improvement
- [ ] ⚡ Performance improvement
- [ ] 🧪 Test addition or improvement

## 📋 Changes
<!-- List the key changes made in this PR -->
- 
- 

### 🔗 Related Issues
<!-- Link any related issues using "Fixes #issue_number" or "Closes #issue_number" -->
- Fixes #
- Related to #

## 🧪 Tests Status
- [ ] Tests ran successfully locally?
- [ ] Added more tests? How many?
- [ ] Code coverage percentage of the codebase: __%

## 📸 Screenshots/Videos
<!-- Add screenshots or screen recordings of UI changes -->
<!-- Use the format below for before/after comparisons -->

| Before | After |
|--------|-------|
|<!-- Add screenshot/video of current behavior -->|<!-- Add screenshot/video of new behavior -->|

## 🎯 Reviewers
<!-- Tag specific reviewers or teams -->
@kcw-grunt, @josikie

kcw-grunt committed:
Fixed the Git Submodule references

kcw-grunt committed:
updated gitmodules config

kcw-grunt committed:
create private gp submodule

kcw-grunt committed:
Update settings.gradle.kts

kcw-grunt commented:
@JosephSanjaya . Just need some help with compiling.

JosephSanjaya commented:
Sorry yesterday already sleep will look into it today  you can leave it to me

JosephSanjaya commented:
just need small chnages from detekt and it can compiled nicely thanks @kcw-grunt

josikie commented:
👀

kcw-grunt commented:
> just need small chnages from detekt and it can compiled nicely thanks @kcw-grunt
Ok thanks @JosephSanjaya  will you push the fix here? Onto this branch?

JosephSanjaya commented:
yes i will push it asap after coming back from office is that okay?

kcw-grunt commented:
> yes i will push it asap after coming back from office is that okay?

Sure. No rush. Thanks for your help @JosephSanjaya

JosephSanjaya committed:
chore: Migrate build logic to version catalog and update submodules

This commit migrates the project's build logic to use a centralized version catalog (`grunt`) and updates submodule references.

Specific changes include:

*   **Submodule Updates:**
    *   Updated `gruntsoftware-build-logic` submodule pointer.
    *   Updated `modules/private-general-purpose` submodule pointer.
*   **Build Configuration (`settings.gradle.kts`):**
    *   Removed the conditional loading of the `games` version catalog.
    *   Added a `grunt` version catalog loaded from `gruntsoftware-build-logic/gradle/libs.versions.toml`.
    *   Changed the included module from `:modules:private-general-purpose:content` to `:modules:private-general-purpose:games`.
    *   Removed conditional inclusion logic for `:modules:private-general-purpose:content`.
*   **Root `build.gradle.kts`:**
    *   Updated plugin aliases from `libs.plugins.android.*` to `grunt.plugins.android.*`.
*   **App `build.gradle.kts`:**
    *   Updated plugin alias for `com.android.application` from `libs` to `grunt`.
    *   Updated the `games` module dependency to `implementation(project(":modules:private-general-purpose:games"))` and removed the conditional logic for `findProject(":modules:games:content")`.
    *   Updated various dependencies to use the `grunt` version catalog (e.g., `grunt.androidx.core.ktx`, `grunt.androidx.compose.bom`, `grunt.bundles.androidx.compose`, `grunt.material`).
*   **Version Catalog (`gradle/libs.versions.toml`):**
    *   Removed versions for `agp` and `androidx-core-ktx`.
    *   Removed versions for `androidx-compose-bom`, `androidx-activity-compose`, and `androidx-fragment-compose`.
    *   Removed `androidx-core` library alias.
    *   Removed various `androidx-compose` library aliases and the `androidx-compose` bundle.
    *   Removed `android-application` and `android-library` plugin aliases.

JosephSanjaya committed:
ci: Configure GitHub token for submodules in CircleCI

This commit updates the CircleCI configuration to include a step that configures a GitHub token. This token will be used for authenticating access to submodules during the build process.

JosephSanjaya commented:
FIxed the CI! @kcw-grunt 

and the changes is already final, waiting for https://github.com/gruntsoftware/android-private-general-purpose/pull/6 to be merged and we can change the git hash to main branches.

kcw-grunt commented:
> FIxed the CI! @kcw-grunt
> 
> and the changes is already final, waiting for [gruntsoftware/android-private-general-purpose#6](https://github.com/gruntsoftware/android-private-general-purpose/pull/6) to be merged and we can change the git hash to main branches.

Thanks @JosephSanjaya I set the Agent for the auth/GHToken in CircleCI

JosephSanjaya committed:
refactor: Update module paths for 'games'

This commit refactors the project structure by moving the 'games' module.
- The 'games' module path in `settings.gradle.kts` is changed from `:modules:private-general-purpose:games` to `:games` and its `projectDir` is set to `modules/private-general-purpose/games`.
- The dependency path in `app/build.gradle.kts` for the 'games' module is updated from `project(":modules:private-general-purpose:games")` to `project(":games")`.

The submodule references for `gruntsoftware-build-logic` and `modules/private-general-purpose` were also updated.

JosephSanjaya reviewed (approved):
LGTM!

josikie reviewed (approved):
Awesome, thank you! @kcw-grunt
